### PR TITLE
Show the choose entity type form in a modal

### PR DESCRIPTION
**Pull request by MKodde**
See individual commits for more details.

This PR should be merged into #174

### CODE REVIEW

**Comment by MKodde** [comment]
Running ant locally indicated no QA issues. But please prove me wrong!

File: app/js/components/modal.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict';
```
Comment (pablothedude) [inline app/js/components/modal.js]: No typescript?

File: docs/third_party_packages.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Use of third party frontend packages
```
Comment (pablothedude) [inline docs/third_party_packages.md]: This is a very valuable addition! Very nice! Thanks!

File: app/js/components/modal.js (hunk 1 above)
Comment (MKodde) [reply app/js/components/modal.js]: Can't get away with this, now can I? ;)

**Comment by MKodde** [comment]
Merging, please continue your review on #174 @pablothedude

**Pull request merged**